[sival/aes] Add AES Interrupt Encryption tests

rprakas-gsc opened the pull request:
@moidx  Checking in aes_interrupt_encryption_test.c code and modified the BUILD to include this test.

luismarques commented on sw/device/tests/aes_interrupt_encryption_test.c:
```diff
@@ -0,0 +1,260 @@
+  LOG_INFO("Started AES trial Encrypt");
+  CHECK_DIF_OK(dif_aes_end(&aes2));
+
+  /** Not checking successful encryption here since its a trial transaction
+   *  Trial encryption ends here
+   *  Resume the Original Encryption below
+   */
```
Nit: why is this using documentation-style comments?

rprakas-gsc replied:
Fixed these comments with regular-style

luismarques commented on sw/device/tests/BUILD:
```diff
@@ -6279,3 +6279,24 @@ opentitan_test(
+    name = "aes_interrupt_encryption_test",
+    srcs = ["aes_interrupt_encryption_test.c"],
+    exec_env = {
+        #         "//hw/top_earlgrey:sim_verilator": None,
+        "//hw/top_earlgrey:fpga_cw340_sival": None,
+        #         "//hw/top_earlgrey:fpga_cw340_rom_with_fake_keys": None,
```
Please clean up the unsupported execution environments. Though perhaps you'd be able to add back verilator, etc?

rprakas-gsc replied:
Done!

luismarques commented on sw/device/tests/aes_interrupt_encryption_test.c:
```diff
@@ -0,0 +1,260 @@
+  CHECK_DIF_OK(dif_aes_trigger(&aes, kDifAesTriggerDataOutClear));
+  CHECK_DIF_OK(dif_aes_trigger(&aes, kDifAesTriggerKeyIvDataInClear));
+
+  LOG_INFO(" De-initialized Automatic operation ");
```
Nit: why the extra whitespace in the string?

rprakas-gsc replied:
Fixed it - and amended the changes!

luismarques commented on sw/device/tests/aes_interrupt_encryption_test.c:
```diff
@@ -0,0 +1,260 @@
+         sizeof(kAesModesPlainText_block34));
+
+  CHECK_DIF_OK(dif_aes_get_status(&aes, kDifAesStatusInputReady, &ready));
+  LOG_INFO("Resuming Encrypt with the remaining 2 blocks of data ");
```
Nit: ditto extraneous whitespace

rprakas-gsc replied:
Fixed this in the updated aes_interrupt_encryption_Test.c code

luismarques reviewed (commented):
Can you add review / commit comments explaining what overall scenario this test tests? I don't suppose there's a testplan testpoint to reference?

engdoreis commented on sw/device/tests/aes_interrupt_encryption_test.c:
```diff
@@ -0,0 +1,260 @@
+
+OTTF_DEFINE_TEST_CONFIG();
+
+status_t execute_test(void) {
```
As this function returns `status_t`, you can replace all instances of `CHECK_DIF_OK` by `TRY`, which would make the code cleaner in my opinion.

rprakas-gsc replied:
Ideally, I want the test code to report FAILURE & abort if the value doesn't get readout correctly , hence using `CHECK_DIF_OK`

engdoreis replied:
The final result would be the same, `TRY` would return immediately and then the callee would abort by using `CHECK_STATUS_OK`.

engdoreis commented on sw/device/tests/aes_interrupt_encryption_test.c:
```diff
@@ -0,0 +1,260 @@
+// manually. Software is expected to get the key in two shares right from the
+// beginning.
+
+static const unsigned char kAesModesPlainText_block12[32] = {
```
The name convention for `const` variables.
```suggestion
static const unsigned char kAesModesPlainTextBlock12[32] = {
```
The same would apply throughout the file.

rprakas-gsc commented:
> Can you add review / commit comments explaining what overall scenario this test tests? I don't suppose there's a testplan testpoint to reference?
There is.  I have added the below lines in the new Commit 
Added AES Interrupt Encryption tests as per test plan identified here:[[…](https://opentitan.org/book/hw/top_earlgrey/data/chip_testplan.html?highlight=chip_sw_aes_interrupt_encryption#chip_sw_aes_interrupt_encryption)
  and modified BUILD

moidx commented:
Hi @nasahlpa, can you take a look at the test implementation to ensure it matches the intended test point description? Thanks!

nasahlpa commented:
> > Can you add review / commit comments explaining what overall scenario this test tests? I don't suppose there's a testplan testpoint to reference?
> > There is.  I have added the below lines in the new Commit
> > Added AES Interrupt Encryption tests as per test plan identified here:[[…](https://opentitan.org/book/hw/top_earlgrey/data/chip_testplan.html?highlight=chip_sw_aes_interrupt_encryption#chip_sw_aes_interrupt_encryption)
> > and modified BUILD

Could you please add the description to the git commit itself? Currently, no description is available in the git commit message. Furthermore, the contributing guideline also mentions that a [subject tag](https://opentitan.org/book/doc/contributing/detailed_contribution_guide/index.html#subject-tags) should be added to the git commit message.

rprakas-gsc commented:
> > > Can you add review / commit comments explaining what overall scenario this test tests? I don't suppose there's a testplan testpoint to reference?
> > > There is.  I have added the below lines in the new Commit
> > > Added AES Interrupt Encryption tests as per test plan identified here:[[…](https://opentitan.org/book/hw/top_earlgrey/data/chip_testplan.html?highlight=chip_sw_aes_interrupt_encryption#chip_sw_aes_interrupt_encryption)
> > > and modified BUILD
> 
> Could you please add the description to the git commit itself? Currently, no description is available in the git commit message. Furthermore, the contributing guideline also mentions that a [subject tag](https://opentitan.org/book/doc/contributing/detailed_contribution_guide/index.html#subject-tags) should be added to the git commit message.

The git commit has restrictions on word limit, if I include the entire link, i will  run out of the world limit and won't be able to commit successfully.

nasahlpa commented:
Understood. But you could add something like this:
```
This commit implements the chip_sw_aes_interrupt_encryption test that is defined in the chip testplan. The purpose of this test is to check, whether an AES encryption can be interrupted and restored.
```
For the header of the git commit message you could use:
```
[sival/aes] Add AES interrupt encryption test
```

rprakas-gsc commented:
Thanks Pascal.  I will incorporate these changes in the commit message now!


On Tue, Oct 8, 2024 at 11:24 PM Pascal Nasahl ***@***.***>
wrote:

> Understood. But you could add something like this:
>
> This commit implements the chip_sw_aes_interrupt_encryption test that is defined in the chip testplan. The purpose of this test is to check, whether an AES encryption can be interrupted and restored.
>
> For the header of the git commit message you could use:
>
> [sival/aes] Add AES interrupt encryption test
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/lowRISC/opentitan/pull/24572#issuecomment-2401417125>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BJEK24Y6LO2262A62JNS7SLZ2TD3VAVCNFSM6AAAAABOFTOFTOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIMBRGQYTOMJSGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

rprakas-gsc commented:
Just updated the PR with the commit messages as suggested!

nasahlpa commented:
Thanks! Would you mind looking into the review comments I've added? Especially the question regarding the decryption should be addressed.

moidx commented:
Hi @nasahlpa, I wasn't able to find your comments. Can you double check if they were sent? Thanks

nasahlpa commented:
> Hi @nasahlpa, I wasn't able to find your comments. Can you double check if they were sent? Thanks

Oh sorry, now the comments should be public.

engdoreis commented on sw/device/tests/BUILD:
Could you please re-order the commit message? The  `Signed-off` line should be after the description.

engdoreis commented on sw/device/tests/aes_interrupt_encryption_test.c:
I suggest the following commit message:
[test, aes] Clean-up log message.

nasahlpa commented on sw/device/tests/aes_interrupt_encryption_test.c:
```diff
@@ -0,0 +1,261 @@
+    memcpy(iv.iv, &iv_encrypt1, sizeof(iv.iv));
+    transaction.manual_operation = kDifAesManualOperationAuto;
+
+    // "Convert" key share byte arrays to `dif_aes_key_share_t`.
```
I think this is not necessary as `key` should still contain the key used for the first two blocks?

nasahlpa reviewed (approved):
Thanks @rprakas-gsc for changing the code based on the feedback. Apart from a small nit this PR looks good to me.

Maybe @engdoreis could have another look.

engdoreis reviewed (approved):
Thanks for the changes.

jwnrt commented:
@rprakas-gsc sorry we missed this in review, but would you mind filing a PR that assigns this test to the testpoint in the testplan please?

https://github.com/lowRISC/opentitan/blob/3a4d766b45e58af05dfccd81a861927724a4783f/hw/top_earlgrey/data/ip/chip_aes_testplan.hjson#L115

i.e.:
```
bazel: ["//sw/device/tests/aes_interrupt_encryption_test"]
```

Feel free to add me as a reviewer

rprakas-gsc commented:
Thanks James! I have create a separate PR request (https://github.com/lowRISC/opentitan/pull/25501) for this.
